lumicks/pylake/kymotracker/detail/trace_line_2d.py

```diff
@@ -188,6 +188,29 @@ def in_rect(self, rect):
         coord_match = np.logical_and(coordinate < rect[1][1], coordinate >= rect[0][1])
         return np.any(np.logical_and(time_match, coord_match))
 
+    def sample_from_image(self, data, n_pixels, reduce=np.sum):
+        """Sample from image using coordinates from this KymoLine.
+
+        This function samples data from the image given in data based on the points in this KymoLine. It samples
+        from [time, position - n_pixels : position + pixels] and then applies the function sum.
+
+        Parameters
+        ----------
+        data : array_like
+            Image to sample from using this KymoLine.
+        n_pixels : int
+            Pixels in either direction to include in the sample
+        reduce : callable
+            Function evaluated on the sample. (Default: np.sum which produces sum of photon counts).
+        """
+        y_shape = data.shape[1]
+
+        return [
+            reduce(data[max(int(coordinate) - n_pixels, 0):min(int(coordinate) + n_pixels + 1, y_shape),
+                   int(time_point)])
+            for time_point, coordinate in zip(self.time, self.coordinate)
+        ]
```

Comment:
1. I feel like `__init__()` should guarantee that `self.time` and `self.coordinates` are arrays of integers if they need to be integers. The extra `int()` casts here are confusing.

2. For list comprehensions I think it's fine to use very short variable names since the parent sequence is right there. E.g.:
```py
        return [reduce(data[max(c - n_pixels, 0):min(c + n_pixels + 1, y_size), t])
                for t, c in zip(self.time, self.coordinate)]
```

Reply:
They shouldn't be integers elsewhere. The positions are subpixel accurate and this is needed for MSD measurements.

They should only be integers here since they're being used to sample an image. Alternatively, one option is to set up an interpolating function with nearest for the 2D data, but I figured that'd even be more confusing.